invoices_app/tests/test_invoice.py

```diff
@@ -430,3 +430,60 @@ def test_invoices_list_filter_invoice_date(self):
             response,
             self.invoice.invoice_number
         )
+
+    def test_invoices_list_filter_invoice_total_amount(self):
+        self.client.force_login(self.ap_user)
+        invoice_low_total_amount = InvoiceFactory(
+            user=self.user,
+            taxpayer=self.taxpayer,
+            invoice_number='4321',
+            total_amount=1,
+        )
+        invoice_high_total_amount = InvoiceFactory(
+            user=self.user,
+            taxpayer=self.taxpayer,
+            invoice_number='98764',
+            total_amount=100,
+        )
+        self.invoice.total_amount = 10
+        self.invoice.save()
+
+        response = self.client.get(
+            '{}?{}&{}'.format(
+                reverse('invoices-list'),
+                'total_amount_min={}'.format(invoice_low_total_amount.total_amount+1),
+                'total_amount_max={}'.format(invoice_high_total_amount.total_amount-1),
+            ),
+        )
+        self.assertNotContains(
+            response,
+            invoice_low_total_amount.invoice_number
+        )
+        self.assertNotContains(
+            response,
+            invoice_high_total_amount.invoice_number
+        )
+        self.assertContains(
+            response,
+            self.invoice.invoice_number
+        )
+
+    def test_invoices_list_filter_taxpayer(self):
+        self.client.force_login(self.ap_user)
+        invoice_other_taxpayer = InvoiceFactory(
+            user=self.user,
+            taxpayer=self.taxpayer_for_other_user,
+            invoice_number='4321'
+        )
```

Comment:
You can user self.invoice_for_other_user in order to reutilize all objects created in TestBase.setUp()

Reply:
Nice!